File: NutritionApp/app/src/main/java/com/bsnutrition/app/core/model/NutritionGoal.kt
package com.bsnutrition.app.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class NutritionGoal(
    val id: Long = 0,
    @SerialName("user_id") val userId: Long = 0,
    @SerialName("effective_from") val effectiveFrom: String? = null,
    @SerialName("calorie_target") val calorieTarget: Int,
    @SerialName("protein_target_g") val proteinTargetG: Float,
    @SerialName("carbohydrate_target_g") val carbohydrateTargetG: Float,
    @SerialName("fat_target_g") val fatTargetG: Float,
    @SerialName("fiber_target_g") val fiberTargetG: Float? = null,
    @SerialName("water_target_ml") val waterTargetMl: Int? = null,
    val source: String = "calculated",
    @SerialName("calculation_version") val calculationVersion: String = "mifflin_v1.0"
)

File: nutrition-backend/app/Models/NutritionGoal.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class NutritionGoal extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'effective_from',
        'calorie_target',
        'protein_target_g',
        'carbohydrate_target_g',
        'fat_target_g',
        'fiber_target_g',
        'water_target_ml',
        'source',
        'calculation_version',
    ];

    protected $casts = [
        'effective_from' => 'date',
        'calorie_target' => 'integer',
        'protein_target_g' => 'float',
        'carbohydrate_target_g' => 'float',
        'fat_target_g' => 'float',
        'fiber_target_g' => 'float',
        'water_target_ml' => 'integer',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
